File: WEEK 1/C++/Calculate_Miles_Without_Refueling.cpp
#include <iostream>
using namespace std;

int main() {
    double capacity, mpg;

    cout << "Enter fuel tank capacity (in gallons): ";
    cin >> capacity;

    cout << "Enter miles per gallon (MPG): ";
    cin >> mpg;

    double milesWithoutRefueling = capacity * mpg;
    printf("The automobile can be driven %.2f miles without refueling.\n", milesWithoutRefueling);

    return 0;
}

File: WEEK 1/JavaScript/Calculate_Miles_Without_Refueling.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function getUserInput(prompt) {
  return new Promise((resolve) => {
    rl.question(prompt, (input) => {
      resolve(parseFloat(input));
    });
  });
}

async function main() {
  try {
    const tankCapacity = await getUserInput('Enter the fuel tank capacity (in gallons): ');
    const mpg = await getUserInput('Enter the miles per gallon (MPG): ');

    if (tankCapacity <= 0 || mpg <= 0) {
      console.log('Both tank capacity and MPG should be positive values.');
      return;
    }

    const milesWithoutRefueling = tankCapacity * mpg;

    console.log(`The automobile can be driven for ${milesWithoutRefueling} miles without refueling.`);
  } catch (error) {
    console.error(error);
  } finally {
    rl.close();
  }
}

main();
